.github/workflows/compliance.yaml

```diff
@@ -0,0 +1,58 @@
+name: Compliance
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
+      - reopened
+      - unassigned
+
+jobs:
+  check-for-linked-issue:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Check for keyword and issue number
+        id: check-for-keyword
+        uses: actions/github-script@v7
+        with:
+          script: |
+            // Retrieve body of context.payload and search for GitHub keywords and issue number
+            const prBody = context.payload.pull_request.body;
+            const prNumber = context.payload.pull_request.number;
+            const prOwner = context.payload.pull_request.user.login;
+            const regex = /(?<!Example:\s*)(close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved)\s+#(\d+)/i;
+            const match = prBody.match(regex);
+            let commentFlag = false;
+            if (match) {
+              let [ , keyword, linkNumber ] = match;
+              console.log('Found a keyword: \'' + keyword + '\'. Checking for legitimate linked issue...');
+              try {
+                const response = await github.rest.issues.get({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  issue_number: linkNumber,
+                });
+                console.log('Found an issue: \'#' + linkNumber + '\' in repo. Confirmed for a legitimate linked issue.');
+              } catch (error) {
+                console.log('Couldn\'t find issue #' + linkNumber + ' in repo. Posting comment...');
+                commentFlag = true;
+              }
+            } else {
+              console.log('PR does not have a properly linked issue. Posting comment...');
+              commentFlag = true;
+            }
+            if (commentFlag) {
```

Comment:
nit. We could refactor this a bit to be a guard clause for simplicity.

Reply:
Refactoring on the next commit